src/lib.rs

```diff
@@ -45,10 +45,18 @@ pub trait EnvIO: Send + Sync + 'static {
 
 #[async_trait::async_trait]
 pub trait HttpIO: Sync + Send + 'static {
+    async fn execute_with<'a>(
+        &'a self,
+        request: reqwest::Request,
+        http_filter: &'a Option<http::HttpFilter>,
```

Comment:
```suggestion
        http_filter: &'a http::HttpFilter,
```
Optional isn't required.